src/main/java/mike/CommandParser.java

```diff
@@ -0,0 +1,293 @@
+
+        consume(TokenType.LITERAL, "Description missing.\n" + usage);
+        String description = previousToken().getText().strip();
+
+        consume(TokenType.FORWARD_DASH, usage);
+        consume(TokenType.PARAM, usage);
+        Token paramToken = previousToken();
+
+        if (!paramToken.getText().equals("from")) {
+            throw error(usage);
+        }
+
+        consume(TokenType.LITERAL, "Start date missing.\n" + usage);
+        String startDate = previousToken().getText().strip();
+
+        consume(TokenType.FORWARD_DASH, usage);
+        consume(TokenType.PARAM, usage);
+        paramToken = previousToken();
+
+        if (!paramToken.getText().equals("to")) {
+            throw error(usage);
+        }
+
+        consume(TokenType.LITERAL, "End date missing.\n" + usage);
+        String endDate = previousToken().getText().strip();
+
+        consume(TokenType.EOC, usage);
+
+        return new AddEventCommand(description, startDate, endDate);
+    }
+
+    private Command parseDelete() throws MikeException {
+        String usage = "Usage: delete [number]";
+
+        consume(TokenType.LITERAL, usage);
+        String argument = previousToken().getText();
+
+        consume(TokenType.EOC, usage);
+
+        try {
+            int taskNumber = Integer.parseInt(argument);
+            return new DeleteCommand(taskNumber);
+        } catch(NumberFormatException e) {
+            String errorMessage = "One, two, three, four, get the kid back through the door!\n" +
+                    "'" + argument + "' is not an integer Sulley...";
+            throw error(errorMessage);
+        }
+    }
+
+    private Command parseFind() throws MikeException {
+        String usage = "Usage: find [keyword]";
+
+        consume(TokenType.LITERAL, usage);
+        String keyword = previousToken().getText().strip();
+
+        consume(TokenType.EOC, usage);
+
+        return new FindCommand(keyword);
+    }
+
+    private boolean match(TokenType...types) {
+        for (TokenType type : types) {
+            if (check(type)) {
+                advance();
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private MikeException error(String message) {
+        return new MikeException(message);
+    }
+
+    private Token consume(TokenType type, String message) throws MikeException {
+        if (check(type)) {
+            return advance();
+        }
+        throw error(message);
+    }
+
+    private boolean check(TokenType type) {
+        return peekToken().getType() == type;
+    }
+
+    private Token advance() {
+        if (!isAtEnd()) {
+            current++;
+        }
+        return previousToken();
+    }
+
+    private Token previousToken() {
+        return tokens.get(current - 1);
+    }
+
+    private boolean isAtEnd() {
+        return peekToken().getType() == TokenType.EOC;
+    }
+
+    private Token peekToken() {
+        return tokens.get(current);
+    }
```

Comment:
the method names specific enough

Reply:
My reasoning was that these methods are private and only meant for use inside the class so it would be specific enough in the context. What do you think?